coova-chilli compile error

When I try to compile coova-chilli in r46409 I get following error.

libtool: compile:  mips-openwrt-linux-musl-gcc -DHAVE_CONFIG_H -I. -I.. -I/trunk/openwrt/staging_dir/target-mips_34kc_musl-1.1.10/usr/include -I/trunk/openwrt/staging_dir/target-mips_34kc_musl-1.1.10/include -I/trunk/openwrt/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.10/usr/include -I/trunk/openwrt/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.10/include/fortify -I/trunk/openwrt/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.10/include -D_GNU_SOURCE -Wall -Werror -fno-builtin -fno-strict-aliasing -O2 -fomit-frame-pointer -funroll-loops -pipe -I../bstring -DDEFCHILLICONF=\"/etc/chilli.conf\" -DDEFPIDFILE=\"/var/run/chilli.pid\" -DDEFSTATEDIR=\"/var/run\" -DSBINDIR=\"/usr/sbin\" -Os -pipe -mno-branch-likely -mips32r2 -mtune=34kc -fno-caller-saves -fhonour-copts -Wno-error=unused-but-set-variable -msoft-float -mips16 -minterlink-mips16 -fstack-protector -Wl,-z,now -Wl,-z,relro -fpic -MT chilli.lo -MD -MP -MF .deps/chilli.Tpo -c chilli.c  -fPIC -DPIC -o .libs/chilli.o
In file included from system.h:139:0,
                 from chilli.h:24,
                 from chilli.c:21:
/trunk/openwrt/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.10/include/sys/sysinfo.h:10:8: error: redefinition of 'struct sysinfo'
 struct sysinfo {
        ^
In file included from system.h:83:0,
                 from chilli.h:24,
                 from chilli.c:21:
/trunk/openwrt/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.10/include/linux/sysinfo.h:7:8: note: originally defined here
 struct sysinfo {
        ^
In file included from /trunk/openwrt/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.10/include/net/ethernet.h:10:0,
                 from system.h:174,
                 from chilli.h:24,
                 from chilli.c:21:
/trunk/openwrt/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.10/include/netinet/if_ether.h:96:8: error: redefinition of 'struct ethhdr'
 struct ethhdr {
        ^
In file included from system.h:130:0,
                 from chilli.h:24,
                 from chilli.c:21:
/trunk/openwrt/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.10/include/linux/if_ether.h:139:8: note: originally defined here
 struct ethhdr {
        ^
In file included from tun.h:26:0,
                 from chilli.h:26,
                 from chilli.c:21:
net.h:171:18: error: 'MAX_SELECT' undeclared here (not in a function)
   select_fd desc[MAX_SELECT];
                  ^
In file included from radius.h:178:0,
                 from chilli.h:28,
                 from chilli.c:21:
radius_pkt.h:28:25: error: 'RADIUS_AUTHLEN' undeclared here (not in a function)
   uint8_t authenticator[RADIUS_AUTHLEN];
                         ^
radius_pkt.h:29:19: error: 'RADIUS_PACKSIZE' undeclared here (not in a function)
   uint8_t payload[RADIUS_PACKSIZE-RADIUS_HDRSIZE];
                   ^
radius_pkt.h:29:35: error: 'RADIUS_HDRSIZE' undeclared here (not in a function)
   uint8_t payload[RADIUS_PACKSIZE-RADIUS_HDRSIZE];
                                   ^
radius_pkt.h:39:16: error: 'RADIUS_ATTR_VLEN' undeclared here (not in a function)
     uint8_t  t[RADIUS_ATTR_VLEN-4];
                ^
In file included from chilli.h:28:0,
                 from chilli.c:21:
radius.h:204:15: error: 'RADIUS_SECRETSIZE' undeclared here (not in a function)
   char secret[RADIUS_SECRETSIZE];
               ^
In file included from redir.h:25:0,
                 from chilli.h:29,
                 from chilli.c:21:
session.h:28:15: error: 'REDIR_USERURLSIZE' undeclared here (not in a function)
   uint8_t url[REDIR_USERURLSIZE];
               ^
session.h:75:17: error: 'REDIR_USERNAMESIZE' undeclared here (not in a function)
   char username[REDIR_USERNAMESIZE];
                 ^
session.h:78:19: error: 'REDIR_MD5LEN' undeclared here (not in a function)
   uint8_t uamchal[REDIR_MD5LEN];
                   ^
session.h:121:18: error: 'REDIR_SESSIONID_LEN' undeclared here (not in a function)
   char sessionid[REDIR_SESSIONID_LEN];
                  ^
In file included from redir.h:26:0,
                 from chilli.h:29,
                 from chilli.c:21:
dhcp.h:198:31: error: 'MAX_RAWIF' undeclared here (not in a function)
   struct _net_interface rawif[MAX_RAWIF];
                               ^
dhcp.h:256:30: error: 'MAX_PASS_THROUGHS' undeclared here (not in a function)
   pass_through pass_throughs[MAX_PASS_THROUGHS];
                              ^
In file included from chilli.h:29:0,
                 from chilli.c:21:
redir.h:97:16: error: 'MAX_EAP_LEN' undeclared here (not in a function)
   uint8_t data[MAX_EAP_LEN];
                ^
redir.h:102:20: error: 'RADIUS_CHAPSIZE' undeclared here (not in a function)
   uint8_t password[RADIUS_CHAPSIZE];
                    ^
redir.h:107:20: error: 'RADIUS_PWSIZE' undeclared here (not in a function)
   uint8_t password[RADIUS_PWSIZE];
                    ^
redir.h:125:19: error: 'REDIR_COOKIESIZE' undeclared here (not in a function)
   char httpcookie[REDIR_COOKIESIZE];   /* Browser Cookies */
                   ^
redir.h:126:13: error: 'REDIR_LANGSIZE' undeclared here (not in a function)
   char lang[REDIR_LANGSIZE];           /* Query string parameter for language */
             ^
In file included from chilli.h:33:0,
                 from chilli.c:21:
options.h:42:13: error: 'OPT_IPADDRLEN' undeclared here (not in a function)
   char netc[OPT_IPADDRLEN];
             ^
options.h:323:20: error: 'MAX_UAM_DOMAINS' undeclared here (not in a function)
   char* uamdomains[MAX_UAM_DOMAINS];
                    ^
options.h:327:17: error: 'MACOK_MAX' undeclared here (not in a function)
   uint8_t macok[MACOK_MAX][PKT_ETH_ALEN]; /* Allowed MACs */
                 ^
In file included from chilli.c:21:0:
chilli.h:114:20: error: 'RADIUS_MPPEKEYSSIZE' undeclared here (not in a function)
   uint8_t lmntkeys[RADIUS_MPPEKEYSSIZE];
                    ^
chilli.h:120:19: error: 'MS2SUCCSIZE' undeclared here (not in a function)
   uint8_t ms2succ[MS2SUCCSIZE];
                   ^
chilli.c: In function 'leaky_bucket_init':
chilli.c:602:43: error: 'BUCKET_TIME' undeclared (first use in this function)
       conn->s_params.bandwidthmaxup / 8 * BUCKET_TIME;
                                           ^
chilli.c:602:43: note: each undeclared identifier is reported only once for each function it appears in
chilli.c:604:38: error: 'BUCKET_SIZE_MIN' undeclared (first use in this function)
     if (conn->s_state.bucketupsize < BUCKET_SIZE_MIN) 
                                      ^
chilli.c: In function 'checkconn':
chilli.c:1192:19: error: 'CHECK_INTERVAL' undeclared (first use in this function)
   if (checkdiff < CHECK_INTERVAL)
                   ^
chilli.c: In function 'chilli_req_attrs':
chilli.c:1287:12: error: 'MACSTRLEN' undeclared (first use in this function)
   char mac[MACSTRLEN+1];
            ^
chilli.c:1287:8: error: unused variable 'mac' [-Werror=unused-variable]
   char mac[MACSTRLEN+1];
        ^
chilli.c: In function 'chilli_auth_radius':
chilli.c:1526:19: error: 'RADIUS_MD5LEN' undeclared (first use in this function)
    0, 0, 0, NULL, RADIUS_MD5LEN);
                   ^
chilli.c: In function 'auth_radius':
chilli.c:1536:12: error: 'MACSTRLEN' undeclared (first use in this function)
   char mac[MACSTRLEN+1];
            ^
chilli.c:1556:56: error: 'USERNAMESIZE' undeclared (first use in this function)
     safe_strncpy(appconn->s_state.redir.username, mac, USERNAMESIZE);
                                                        ^
chilli.c:1621:19: error: 'RADIUS_MD5LEN' undeclared (first use in this function)
    0, 0, 0, NULL, RADIUS_MD5LEN);
                   ^
chilli.c:1536:8: error: unused variable 'mac' [-Werror=unused-variable]
   char mac[MACSTRLEN+1];
        ^
chilli.c: In function 'radius_access_challenge':
chilli.c:1689:19: error: 'RADIUS_MD5LEN' undeclared (first use in this function)
    0, 0, 0, NULL, RADIUS_MD5LEN);
                   ^
chilli.c: In function 'radius_access_accept':
chilli.c:1751:19: error: 'RADIUS_MD5LEN' undeclared (first use in this function)
    0, 0, 0, NULL, RADIUS_MD5LEN);
                   ^
chilli.c:1702:11: error: unused variable 'mppekey' [-Werror=unused-variable]
   uint8_t mppekey[RADIUS_ATTR_VLEN];
           ^
chilli.c: In function 'dnprot_reject':
chilli.c:2123:58: error: 'USERNAMESIZE' undeclared (first use in this function)
       safe_strncpy(appconn->s_state.redir.username, "-", USERNAMESIZE);
                                                          ^
In file included from tun.h:25:0,
                 from chilli.h:26,
                 from chilli.c:21:
chilli.c: In function 'cb_tun_ind':
pkt.h:79:20: error: 'PKT_MAX_LEN' undeclared (first use in this function)
 #define PKT_BUFFER PKT_MAX_LEN
                    ^
chilli.c:2522:17: note: in expansion of macro 'PKT_BUFFER'
  uint8_t packet[PKT_BUFFER];
                 ^
chilli.c:2522:10: error: unused variable 'packet' [-Werror=unused-variable]
  uint8_t packet[PKT_BUFFER];
          ^
chilli.c: In function 'accounting_request':
chilli.c:3159:8: error: unused variable 'macstr' [-Werror=unused-variable]
   char macstr[RADIUS_ATTR_VLEN];
        ^
chilli.c: In function 'access_request':
chilli.c:3612:22: error: 'USERNAMESIZE' undeclared (first use in this function)
   if (uidattr->l-2 < USERNAMESIZE) {
                      ^
chilli.c:3677:19: error: 'RADIUS_MD5LEN' undeclared (first use in this function)
    0, 0, 0, NULL, RADIUS_MD5LEN);
                   ^
chilli.c:3381:11: error: unused variable 'resp' [-Werror=unused-variable]
   uint8_t resp[MAX_EAP_LEN];     /* EAP response */
           ^
chilli.c:3373:8: error: unused variable 'macstr' [-Werror=unused-variable]
   char macstr[RADIUS_ATTR_VLEN];
        ^
chilli.c:3370:8: error: unused variable 'pwd' [-Werror=unused-variable]
   char pwd[RADIUS_ATTR_VLEN];
        ^
chilli.c: In function 'config_radius_session':
chilli.c:4160:10: error: unused variable 'attrs' [-Werror=unused-variable]
     char attrs[RADIUS_ATTR_VLEN + 1];
          ^
chilli.c: In function 'cb_dhcp_request':
chilli.c:4956:13: error: 'MACSTRLEN' undeclared (first use in this function)
    char mac[MACSTRLEN+1];
             ^
chilli.c:4960:55: error: 'USERNAMESIZE' undeclared (first use in this function)
    safe_strncpy(appconn->s_state.redir.username, mac, USERNAMESIZE);
                                                       ^
chilli.c:4956:9: error: unused variable 'mac' [-Werror=unused-variable]
    char mac[MACSTRLEN+1];
         ^
chilli.c: In function 'chilli_cmd':
chilli.c:6831:17: error: 'USERNAMESIZE' undeclared (first use in this function)
          uname, USERNAMESIZE);
                 ^
cc1: all warnings being treated as errors
make[7]: *** [chilli.lo] Error 1
make[7]: Leaving directory `/trunk/openwrt/build_dir/target-mips_34kc_musl-1.1.10/coova-chilli-1.3.0+20141128/src'
make[6]: *** [all-recursive] Error 1
make[6]: Leaving directory `/trunk/openwrt/build_dir/target-mips_34kc_musl-1.1.10/coova-chilli-1.3.0+20141128/src'
make[5]: *** [all-recursive] Error 1
make[5]: Leaving directory `/trunk/openwrt/build_dir/target-mips_34kc_musl-1.1.10/coova-chilli-1.3.0+20141128'
make[4]: *** [all] Error 2
make[4]: Leaving directory `/trunk/openwrt/build_dir/target-mips_34kc_musl-1.1.10/coova-chilli-1.3.0+20141128'
make[3]: *** [/trunk/openwrt/build_dir/target-mips_34kc_musl-1.1.10/coova-chilli-1.3.0+20141128/.built] Error 2
make[3]: Leaving directory `/trunk/openwrt/feeds/packages/net/coova-chilli'
make[2]: *** [package/feeds/packages/coova-chilli/compile] Error 2
make[2]: Leaving directory `/trunk/openwrt'
make[1]: *** [/trunk/openwrt/staging_dir/target-mips_34kc_musl-1.1.10/stamp/.package_compile] Error 2
make[1]: Leaving directory `/trunk/openwrt'
make: *** [world] Error 2

--- a/net/coova-chilli/patches/400-fix-compile-with-musl.patch
+++ b/net/coova-chilli/patches/400-fix-compile-with-musl.patch
@@ -1,0 +1,61 @@
+--- a/src/system.h
++++ b/src/system.h
+@@ -78,11 +78,6 @@
+ #include <sys/stat.h>
+ #endif
+ 
+-#ifdef HAVE_LINUX_SYSINFO_H
+-#define _LINUX_KERNEL_H
+-#include <linux/sysinfo.h>
+-#endif
+-
+ #ifdef HAVE_TIME_H
+ #include <time.h>
+ #endif
+@@ -123,6 +118,11 @@
+ #include <signal.h>
+ #endif
+ 
++#ifdef HAVE_SYS_SYSINFO_H
++#define _LINUX_SYSINFO_H
++#include <sys/sysinfo.h>
++#endif
++
+ #if defined(__linux__)
+ #include <asm/types.h>
+ #include <linux/if.h>
+@@ -135,10 +135,6 @@
+ #include <linux/un.h>
+ #endif
+ 
+-#ifdef HAVE_SYS_SYSINFO_H
+-#include <sys/sysinfo.h>
+-#endif
+-
+ #elif defined (__FreeBSD__)  || defined (__APPLE__) || defined (__OpenBSD__) || defined (__NetBSD__) 
+ #include <net/if.h>
+ #include <net/bpf.h>
+@@ -170,10 +166,6 @@
+ #include <net/if_tun.h>
+ #endif
+ 
+-#ifdef HAVE_NET_ETHERNET_H
+-#include <net/ethernet.h>
+-#endif
+-
+ #ifdef HAVE_ASM_TYPES_H
+ #include <asm/types.h>
+ #endif
+--- a/src/chilli_limits.h
++++ b/src/chilli_limits.h
+@@ -18,8 +18,8 @@
+  * 
+  */
+ 
+-#ifndef _LIMITS_H
+-#define _LIMITS_H
++#ifndef _CHILLI_LIMITS_H
++#define _CHILLI_LIMITS_H
+ 
+ /*
+  * extracted from various .h files, needs some cleanup.

